### PR TITLE
Move ingress-gateway DR to minikube env only

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Move ingress DestinationRule to localenv
- Add migration job for DR deletion
- Update base image for api-gateway tests

**Related issue(s)**

- Implements #7624

### CODE REVIEW

File: resources/istio/files/destination-rules-minikube.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+# DestinationRule to disable (m)TLS when talking to helm tiller, as it doesn't have a sidecar.
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: tiller-deploy
+spec:
+  host: tiller-deploy.kube-system.svc.cluster.local
```
Comment (piotrmsc) [inline resources/istio/files/destination-rules-minikube.yaml:8]: tiller should be applied always :)

File: resources/istio/templates/configmap-extra-manifests.yaml [hunk 2]
```diff
@@ -5,4 +5,8 @@ metadata:
   name: kyma-extra-manifests
 data:
   destination-rules.yaml: |-
-{{tpl (.Files.Get "files/destination-rules.yaml" | printf "%s" | indent 4) .}}
+{{- if .Values.global.isLocalEnv }}
+{{ tpl (.Files.Get "files/destination-rules-minikube.yaml" | printf "%s" | indent 4) . }}
+{{- else }}
```
Comment (piotrmsc) [inline resources/istio/templates/configmap-extra-manifests.yaml:10]: and then this if else can be simplified to single if containing additional DR for ingress :)

File: resources/istio/templates/rbac-migration-remove-dr.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+apiVersion: v1
+kind: ServiceAccount
```
Comment (piotrmsc) [inline resources/istio/templates/rbac-migration-remove-dr.yaml:3]: Do we need that? this DR will be removed on upgrade by tiller?

File: resources/istio/templates/job-migration-remove-dr.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+apiVersion: batch/v1
+kind: Job
+metadata:
+  annotations:
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    helm.sh/hook: "pre-upgrade"
+    helm.sh/hook-weight: "5"
+  name: remove-dr
+spec:
+  activeDeadlineSeconds: 1000
```
Comment (piotrmsc) [inline resources/istio/templates/job-migration-remove-dr.yaml:11]: See below

File: resources/istio/templates/rbac-migration-remove-dr.yaml (hunk 3 above)
Comment (Demonsthere) [reply resources/istio/templates/rbac-migration-remove-dr.yaml:3]: The DR is not managed by tiller, we create by with kubectl in the install job, so we need to remove it during upgrades

Comment (piotrmsc) [reply resources/istio/templates/rbac-migration-remove-dr.yaml:3]: Yeah, I forgot that this was changed and is applied as a part of istio-installer job :)

File: resources/istio/templates/rbac-migration-remove-dr.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+---
```
Comment (piotrmsc) [inline resources/istio/templates/rbac-migration-remove-dr.yaml:1]: Please create a follow up for us to clean this after next release :)

File: resources/istio/templates/job-migration-remove-dr.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  template:
+    spec:
+      containers:
+        -
+          command:
+            - bash
+            - "-c"
+            - |
+              set -e
+              kubectl delete dr istio-ingressgateway -n istio-system
```
Comment (piotrmsc) [inline resources/istio/templates/job-migration-remove-dr.yaml:22]: nit :  we can prepare ourselves for the future :D by addidn --ignore-not-found  flag. That will prevent upgrade plan from failing after 1.12 before we remove this delete

File: resources/istio/templates/rbac-migration-remove-dr.yaml (hunk 5 above)
Comment (Demonsthere) [reply resources/istio/templates/rbac-migration-remove-dr.yaml:1]: Followup created: https://github.com/kyma-project/kyma/issues/8066

**Pull request merged**